src/sdam/description/server.rs

```diff
@@ -74,7 +73,7 @@ pub(crate) struct ServerDescription {
     // allows us to ensure that only valid states are possible (e.g. preventing that both an error
     // and a reply are present) while still making it easy to define helper methods on
     // ServerDescription for information we need from the isMaster reply by propagating with `?`.
-    pub(crate) reply: Result<Option<IsMasterReply>>,
+    pub(crate) reply: Result<Option<IsMasterReply>, String>,
```

Comment:
Because the only place we use the errors on server descriptions is in the error messages to users when server selection fails, the simplest way to avoid needing to copy errors around here is to just store the description of the error instead of the error itself, since that's what will be shown to users anyhow. The rest of the changes in this specific commit are due to this one change.